internal/component/common/connector.go

```diff
@@ -0,0 +1,127 @@
+package common
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"log/slog"
+
+	"github.com/goccy/go-json"
+	"github.com/goto/optimus-any2any/pkg/component"
+	"github.com/goto/optimus-any2any/pkg/flow"
+	"github.com/pkg/errors"
+)
+
+type ConnectorExecFunc func(inputReader io.Reader) (io.Reader, error)
+
+// Connector is a struct that extends the component.Connector
+// to provide additional functionality for processing records in batches
+// with some specialized data handling.
+type Connector struct {
+	*component.Connector
+	metadataPrefix   string
+	batchSize        int
+	batchIndexColumn string
+	exec             ConnectorExecFunc
+}
+
+func NewConnector(ctx context.Context, cancelFn context.CancelCauseFunc, logger *slog.Logger, metadataPrefix string, batchSize int, batchIndexColumn string, name string) (*Connector, error) {
+	c := &Connector{
+		Connector:        component.NewConnector(ctx, cancelFn, logger, name),
+		metadataPrefix:   metadataPrefix,
+		batchSize:        batchSize,
+		batchIndexColumn: batchIndexColumn,
+	}
+	c.Connector.SetConnectorFunc(c.process)
+	return c, nil
+}
+
+// SetExecFunc sets the execution function for the Connector.
+func (c *Connector) SetExecFunc(execFunc ConnectorExecFunc) {
+	c.exec = execFunc
+}
```

Comment:
since setting execFunc is now outside of the **New** function, shall we add check in `process` to quit when there's no exec func set? Preventing panic due to nil pointer